contracts/dezswap_pair/src/contract.rs

```diff
@@ -361,6 +371,12 @@ pub fn provide_liquidity(
                 funds: vec![],
             }));
         }
+
+        if let Some(slippage_tolerance) = slippage_tolerance {
+            if remain_amount > desired_amount * slippage_tolerance {
```

Comment:
I think we compare deposits[i] not desired_amount.

Reply:
https://github.com/dezswap/dezswap-contracts/pull/14/commits/f20b92b78f841409454f5aa87cf4411eeaf25db5